src/rpc/rawtransaction.cpp

```diff
@@ -218,6 +218,252 @@ UniValue getrawtransaction(const UniValue &params, bool fHelp)
+            "         \"scriptSig\": {     (json object) The script\n"
+            "           \"asm\": \"asm\",  (string) asm\n"
+            "           \"hex\": \"hex\"   (string) hex\n"
+            "         },\n"
+            "         \"sequence\": n      (numeric) The script sequence number\n"
+            "       }\n"
+            "       ,...\n"
+            "      ],\n"
+            "    \"vout\" : [              (array of json objects)\n"
+            "       {\n"
+            "         \"value\" : x.xxx,            (numeric) The value in " +
+            CURRENCY_UNIT +
+            "\n"
+            "         \"n\" : n,                    (numeric) index\n"
+            "         \"scriptPubKey\" : {          (json object)\n"
+            "           \"asm\" : \"asm\",          (string) the asm\n"
+            "           \"hex\" : \"hex\",          (string) the hex\n"
+            "           \"reqSigs\" : n,            (numeric) The required sigs\n"
+            "           \"type\" : \"pubkeyhash\",  (string) The type, eg 'pubkeyhash'\n"
+            "           \"addresses\" : [           (json array of string)\n"
+            "             \"bitcoinaddress\"        (string) bitcoin address\n"
+            "             ,...\n"
+            "           ]\n"
+            "         }\n"
+            "       }\n"
+            "      ,...\n"
+            "      ],\n"
+            "    \"blockhash\" : \"hash\",   (string) the block hash\n"
+            "    \"confirmations\" : n,      (numeric) The confirmations\n"
+            "    \"time\" : ttt,             (numeric) The transaction time in seconds since epoch (Jan 1 1970 GMT)\n"
+            "    \"blocktime\" : ttt         (numeric) The block time in seconds since epoch (Jan 1 1970 GMT)\n"
+            "  },\n"
+            "  ...\n"
+            "}\n"
+            "\nExamples:\n" +
+            HelpExampleCli("getrawtransactions", "\"hashblock\"") +
+            HelpExampleCli("getrawtransactions", "\"hashblock\" 1") +
+            HelpExampleRpc("getrawtransactions", "\"hashblock\", 1"));
+
+    uint256 hashBlock = ParseHashV(params[0], "parameter 1");
+
+    bool fVerbose = false;
+    if (params.size() > 1)
+        fVerbose = (params[1].get_int() != 0);
+
+    CBlockIndex *pblockindex = nullptr;
+    pblockindex = LookupBlockIndex(hashBlock);
+    if (!pblockindex)
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
+
+    CBlock block;
+    if (!ReadBlockFromDisk(block, pblockindex, Params().GetConsensus()))
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Can't read block from disk");
+
+    UniValue resultSet(UniValue::VOBJ);
+    for (auto tx : block.vtx)
+    {
+        string strHex = EncodeHexTx(*tx);
+
+        if (!fVerbose)
+        {
+            resultSet.pushKV(tx->GetHash().GetHex(), strHex);
+            continue;
+        }
+
+        UniValue result(UniValue::VOBJ);
+        result.pushKV("hex", strHex);
```

Comment:
What about not providing the hex if verbose is set?  Why would someone want hex and verbose, and is this common enough to optimize it as opposed to having the user call the API 2x? 

If we drop the hex, this will speed things up because the message will be smaller and there will be less JSON encode/decode.

Reply:
I left the hex in there to be consistent with getrawtransaction which also returns the hex regardless of verbosity. i have no problem removing it from all of the calls but i feel some feedback from app devs should be collected first.